### PR TITLE
TS: Add timestamp to IRequestHandler

### PR DESCRIPTION
**Pull request by f-squirrel**
* Updated all implementation of IRequestHandler::execute.
* Added logic to calculate the time offset for single requests within a
batch

Don't be afraid to review it, most of the changes is just an update for all the classes deriving from `IRequestHandler` :)

### CODE REVIEW

**Review by tdimitrov — APPROVED** [review]
Looks good to me. I've got one question only (as comment).
Good job @f-squirrel

File: bftengine/src/preprocessor/PreProcessor.cpp [hunk 1]
```diff
@@ -1120,7 +1121,7 @@ uint32_t PreProcessor::launchReqPreProcessing(uint16_t clientId,
       std::move(signature),
       maxPreExecResultSize_,
       (char *)getPreProcessResultBuffer(clientId, reqSeqNum, reqOffsetInBatch)});
-  requestsHandler_.execute(accumulatedRequests, cid, span);
+  requestsHandler_.execute(accumulatedRequests, std::nullopt, cid, span);
```
Comment (tdimitrov) [inline bftengine/src/preprocessor/PreProcessor.cpp:1124]: Just curious - why the timestamp is not set in preexecution?

Comment (f-squirrel) [reply bftengine/src/preprocessor/PreProcessor.cpp:1124]: The reason is because in pre-execution we cannot guarantee full consensus on the timestamp.

**Review by glevkovich — APPROVED** [review]
LGTM

File: bftengine/src/preprocessor/PreProcessor.cpp (hunk 1 above)
Comment (tdimitrov) [reply bftengine/src/preprocessor/PreProcessor.cpp:1124]: Makes sense. Thanks!

**Pull request merged**